### PR TITLE
Use typescript for server code

### PR DESCRIPTION
**Pull request by bluquar**
http://chris.dandelion.supplies is up and running

### CODE REVIEW

File: .github/workflows/android-build.yaml [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
+    name: Android
     runs-on: ubuntu-latest
     environment: EAS
     steps:
       - name: Checkout source code
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Exit early if client was not modified
```
Comment (lowellbander) [inline .github/workflows/android-build.yaml:14]: very smart!

File: App.tsx [hunk 2]
```diff
@@ -1,23 +1,3 @@
...
-  } else {
-    return (
-      <SafeAreaProvider>
-        <Navigation colorScheme={colorScheme} />
-        <StatusBar />
-      </SafeAreaProvider>
-    );
-  }
-}
+export default App;
```
Comment (lowellbander) [inline App.tsx:3]: yay thin top level!

File: package.json [hunk 3]
```diff
@@ -3,26 +3,28 @@
   "version": "0.0.0",
   "main": "node_modules/expo/AppEntry.js",
   "scripts": {
-    "start": "node ./src/bin/www",
-    "dev:web": "yarn install && npm run expo:build:web && npm start",
+    "start": "node server-build/runserver.js",
```
Comment (lowellbander) [inline package.json:6]: yay file extensions!

File: src/client/hooks/useCachedResources.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ export default function useCachedResources() {
         // Load fonts
         await Font.loadAsync({
           ...FontAwesome.font,
-          'space-mono': require('../assets/fonts/SpaceMono-Regular.ttf'),
+          'space-mono': require('../../../assets/fonts/SpaceMono-Regular.ttf'),
```
Comment (lowellbander) [inline src/client/hooks/useCachedResources.ts:18]: lul

**Review by lowellbander — APPROVED** [review]
whoa why 22 conversation entries for a single commit?

you're so good at this! you're creating something really cool!

do the files in `server-build/` need to be tracked by the repo in order to be deployed? that seems like something that shouldn't be tracked and instead built on the fly by the deployment process, right?

**Comment by bluquar** [comment]
Oh totally, I forgot to add server-build to the .gitignore. Good catch!

**Comment by bluquar** [comment]
I edited the branch history to make it into a single commit. There were a lot of intermediate commits that I pushed, which triggered the builds

**Pull request merged**